src/tools/spectest-interp.cc

```diff
@@ -1274,15 +1274,21 @@ wabt::Result CommandRunner::Run(const Script& script) {
   for (const CommandPtr& command : script.commands) {
     switch (command->type) {
       case CommandType::Module:
-        OnModuleCommand(cast<ModuleCommand>(command.get()));
+        if (Failed(OnModuleCommand(cast<ModuleCommand>(command.get())))) {
+          PrintError(command->line, "invalid module");
+          return wabt::Result::Error;
+        }
         break;
 
       case CommandType::Action:
         TallyCommand(OnActionCommand(cast<ActionCommand>(command.get())));
         break;
 
       case CommandType::Register:
-        OnRegisterCommand(cast<RegisterCommand>(command.get()));
+        if (Failed(OnRegisterCommand(cast<RegisterCommand>(command.get())))) {
+          PrintError(command->line, "invalid register command");
+          return wabt::Result::Error;
```

Comment:
I guess register is different because it can/will effect all future commands so hard here makes more sense.